optimism/src/keccak/constraints.rs

```diff
@@ -64,6 +65,47 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
         }
 
         // SPONGE CONSTRAINTS
+        {
+            for z in self.sponge_zeros() {
```

Comment:
Follow-up/reminder: check this twice when the final structure is merged.

Reply:
actually, I realized of an optimization that reduces by 32 constraints in the sponge, I am creating a follow-up branch called `keccak/32-less-constraints`

Reply:
Okay, false alarm. Then I need to check that the other 32 entries are unchanged (xor by zero), that doesnt change things then).